### ISSUE TITLE
[Question] Ping rangeStrategy not working for maven, or bad config?

### ISSUE THREAD

**Issue by jamesmmchugh** [renovatebot/config-help#237]
### Which Renovate are you using? CLI, App, or Pro
OSS docker

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps
GitHub

### What is your question?
I have this config:

```
{
  "extends": [
    "config:base"
  ],
  "maven": {
    "ignoreDeps": ["org.apache.commons:commons-math3"]
  },
  "packageRules": [
    {
      "managers": ["maven"],
      "packageNames": ["com.google.guava:guava"],
      "versionScheme": "docker"
    },
    {
      "managers": ["maven"],
      "packageNames": ["org.projectlombok:lombok"],
      "rangeStrategy": "pin"
    }
  ]
}
```

With this lombok dependency in my maven pom:

```
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>[1.12.6,1.18.6]</version>
        </dependency>
```

And the latest version of lombok being 1.18.8.

Which, if I understand correctly, should pin the lombok dependency version to the version `1.18.8` when it get's updated. However, the onboarding PR states:

`Upgrade org.projectlombok:lombok to [1.12.6,1.18.8]`

Am i misunderstanding, or mis-configuring renovate?

Cheers

**Comment by jamesmmchugh** [comment]
Using Renovate version 17.4.1

**Comment by rarkins** [comment]
Regardless of the package manager, Renovate will always pin to the version *inside* the existing range first, before upgrading it. The goal is that you can fairly safely/quickly adopt Pin PRs as they should result in the same installed dependencies are you got currently from the range. In your case, that's 1.18.6 and not 1.18.8.

However it looks like the pinning is not taking effect here, so you're just getting a regular range update. I would need to check if such a capability is supported by Maven or if we somehow left it out. I think your config looks OK.

**Comment by jamesmmchugh** [comment]
Is it because the latest version is outside the range, so that it will update the range first then when I next run it will pin inside the range? I'll try it manually by updating the range and re-running the onboarding.

**Comment by jamesmmchugh** [comment]
Nope just tried and then it ignores lombok completely.

**Comment by rarkins** [comment]
No, it always does pin first, then upgrade. Any upgrade PR gets suppressed if there is a pin PR for the same

**Comment by jamesmmchugh** [comment]
I see, then yes it appears pinning isn't taking affect for maven. Should i raise this as a bug in the main renovate repo? It should be possible as the maven versions plugin can do this, and maven has to resolve a specific version anyway when building the artefact.

**Comment by rarkins** [comment]
Let me try some magic with this:
![image](https://user-images.githubusercontent.com/6311784/57842764-74050d80-77cd-11e9-925c-8d64ba073dba.png)
